From f740b319c287f03227baadaef96171a8621303da Mon Sep 17 00:00:00 2001
From: Jean Delvare <jdelvare@suse.de>
Date: Tue, 23 Sep 2025 13:42:04 +0200
Subject: [PATCH 39/45] dmidecode: Rename BIOS to Firmware

Starting with SMBIOS specification version 3.8.0, the term "BIOS" is
replaced by the more generic term "Firmware" or "Platform Firmware".
Update all references accordingly.

Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 dmidecode.c     | 32 ++++++++++++++++----------------
 man/dmidecode.8 | 10 +++++-----
 2 files changed, 21 insertions(+), 21 deletions(-)

diff --git a/dmidecode.c b/dmidecode.c
index 5709e8a..5298123 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -162,7 +162,7 @@ const char *dmi_string(const struct dmi_header *dm, u8 s)
 static const char *dmi_smbios_structure_type(u8 code)
 {
 	static const char *type[] = {
-		"BIOS", /* 0 */
+		"Platform Firmware", /* 0 */
 		"System",
 		"Base Board",
 		"Chassis",
@@ -175,7 +175,7 @@ static const char *dmi_smbios_structure_type(u8 code)
 		"On Board Devices",
 		"OEM Strings",
 		"System Configuration Options",
-		"BIOS Language",
+		"Firmware Language",
 		"Group Associations",
 		"System Event Log",
 		"Physical Memory Array",
@@ -340,7 +340,7 @@ void dmi_print_storage_size(const char *attr, u64 code, unsigned int shift)
 }
 
 /*
- * 7.1 BIOS Information (Type 0)
+ * 7.1 Platform Firmware Information (Type 0)
  */
 
 static void dmi_bios_runtime_size(u32 code)
@@ -378,7 +378,7 @@ static void dmi_bios_characteristics(u64 code)
 {
 	/* 7.1.1 */
 	static const char *characteristics[] = {
-		"BIOS characteristics not supported", /* 3 */
+		"Firmware characteristics not supported", /* 3 */
 		"ISA is supported",
 		"MCA is supported",
 		"EISA is supported",
@@ -386,13 +386,13 @@ static void dmi_bios_characteristics(u64 code)
 		"PC Card (PCMCIA) is supported",
 		"PNP is supported",
 		"APM is supported",
-		"BIOS is upgradeable",
-		"BIOS shadowing is allowed",
+		"Firmware is upgradeable",
+		"Firmware shadowing is allowed",
 		"VLB is supported",
 		"ESCD support is available",
 		"Boot from CD is supported",
 		"Selectable boot is supported",
-		"BIOS ROM is socketed",
+		"Firmware ROM is socketed",
 		"Boot from PC Card (PCMCIA) is supported",
 		"EDD is supported",
 		"Japanese floppy for NEC 9800 1.2 MB is supported (int 13h)",
@@ -1398,8 +1398,8 @@ static const char *dmi_processor_status(u8 code)
 	static const char *status[] = {
 		"Unknown", /* 0x00 */
 		"Enabled",
-		"Disabled By User",
-		"Disabled By BIOS",
+		"Disabled by user",
+		"Disabled by firmware",
 		"Idle", /* 0x04 */
 		out_of_spec,
 		out_of_spec,
@@ -2495,7 +2495,7 @@ static void dmi_system_configuration_options(const struct dmi_header *h)
 }
 
 /*
- * 7.14 BIOS Language Information (Type 13)
+ * 7.14 Firmware Language Information (Type 13)
  */
 
 static void dmi_bios_languages(const struct dmi_header *h)
@@ -4473,8 +4473,8 @@ static void dmi_decode(const struct dmi_header *h, u16 ver)
 	 */
 	switch (h->type)
 	{
-		case 0: /* 7.1 BIOS Information */
-			pr_handle_name("BIOS Information");
+		case 0: /* 7.1 Platform Firmware Information */
+			pr_handle_name("Platform Firmware Information");
 			if (h->length < 0x12) break;
 			pr_attr("Vendor", "%s",
 				dmi_string(h, data[0x04]));
@@ -4503,10 +4503,10 @@ static void dmi_decode(const struct dmi_header *h, u16 ver)
 			pr_list_end();
 			if (h->length < 0x18) break;
 			if (data[0x14] != 0xFF && data[0x15] != 0xFF)
-				pr_attr("BIOS Revision", "%u.%u",
+				pr_attr("Platform Firmware Revision", "%u.%u",
 					data[0x14], data[0x15]);
 			if (data[0x16] != 0xFF && data[0x17] != 0xFF)
-				pr_attr("Firmware Revision", "%u.%u",
+				pr_attr("Embedded Controller Firmware Revision", "%u.%u",
 					data[0x16], data[0x17]);
 			break;
 
@@ -4798,8 +4798,8 @@ static void dmi_decode(const struct dmi_header *h, u16 ver)
 			dmi_system_configuration_options(h);
 			break;
 
-		case 13: /* 7.14 BIOS Language Information */
-			pr_handle_name("BIOS Language Information");
+		case 13: /* 7.14 Firmware Language Information */
+			pr_handle_name("Firmware Language Information");
 			if (h->length < 0x16) break;
 			if (ver >= 0x0201)
 			{
diff --git a/man/dmidecode.8 b/man/dmidecode.8
index 6636b72..f25f9a0 100644
--- a/man/dmidecode.8
+++ b/man/dmidecode.8
@@ -1,6 +1,6 @@
 '\" t
 .\" ** The above line should force tbl to be a preprocessor **
-.TH DMIDECODE 8 "February 2023" "dmidecode"
+.TH DMIDECODE 8 "June 2025" "dmidecode"
 .\"
 .SH NAME
 dmidecode \- \s-1DMI\s0 table decoder
@@ -14,7 +14,7 @@ dmidecode \- \s-1DMI\s0 table decoder
 is a tool for dumping a computer's \s-1DMI\s0 (some say \s-1SMBIOS\s0) table
 contents in a human-readable format. This table contains a description of the
 system's hardware components, as well as other useful pieces of information
-such as serial numbers and \s-1BIOS\s0 revision. Thanks to this table, you can
+such as serial numbers and firmware revision. Thanks to this table, you can
 retrieve this information without having to probe for the actual hardware.
 While this is a good point in terms of report speed and safeness, this also
 makes the presented information possibly unreliable.
@@ -203,7 +203,7 @@ determine the output format and are mutually exclusive.
 .P
 Please note in case of
 .B dmidecode
-is run on a system with BIOS that boasts new SMBIOS specification, which
+is run on a system with firmware that boasts new SMBIOS specification, which
 is not supported by the tool yet, it will print out relevant message in
 addition to requested data on the very top of the output. Thus informs the
 output data is not reliable.
@@ -215,7 +215,7 @@ r l
 __
 r l.
 Type	Information
-0	BIOS
+0	Platform Firmware
 1	System
 2	Baseboard
 3	Chassis
@@ -228,7 +228,7 @@ Type	Information
 10	On Board Devices
 11	OEM Strings
 12	System Configuration Options
-13	BIOS Language
+13	Firmware Language
 14	Group Associations
 15	System Event Log
 16	Physical Memory Array
-- 
2.47.0

